server/jetstream_cluster.go

```diff
@@ -4420,6 +4421,20 @@ func (mset *stream) processClusteredInboundMsg(subject, reply string, hdr, msg [
 		return err
 	}
 
+	// Since we encode header len as u16 make sure we do not exceed.
+	// Again this works if it goes through but better to be pre-emptive.
+	if len(hdr) > math.MaxUint16 {
+		err := fmt.Errorf("JetStream header size exceeds limits for '%s > %s'", jsa.acc().Name, mset.cfg.Name)
```

Comment:
on the assumption someone might be trying to trigger overflows or memory exchaustion attacks should we maybe log the IP of the client or some info about him?

Reply:
If the message violates total payload limits or control line limits the error is logged and the client cutoff.

Reply:
Yes, but I am mainly interested in having a way to know who is the bad client - i guess events will be raised so its probably fine. LGTM..